datadog_lambda/asm.py

```diff
@@ -126,6 +130,14 @@ def asm_start_request(
         span.set_tag_str("http.client_ip", request_ip)
         span.set_tag_str("network.client.ip", request_ip)
 
+    # Encode the parsed query and append it to reconstruct the original raw URI expected by AppSec.
+    if parsed_query:
+        try:
+            encoded_query = urllib.parse.urlencode(parsed_query, doseq=True)
+        except Exception:
+            pass
+        raw_uri += "?" + encoded_query  # type: ignore
```

Comment:
Here's one way to better handle this.

```suggestion
        try:
            encoded_query = urllib.parse.urlencode(parsed_query, doseq=True)
            raw_uri += "?" + encoded_query  # type: ignore
        except Exception:
            pass
```